nixos/modules/services/video/v4l2-relayd.nix

```
@@ -0,0 +1,139 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+
+  cfg = config.services.v4l2-relayd;
+
+  ipu6Platform = config.hardware.ipu6.platform;
+
+  gst = (with pkgs.gst_all_1; [
+    gst-plugins-bad
+    gst-plugins-base
+    gst-plugins-good
+    gstreamer.out
+  ]) ++ optional (ipu6Platform == "ipu6") pkgs.gst_all_1.icamerasrc-ipu6
+  ++ optional (ipu6Platform == "ipu6ep") pkgs.gst_all_1.icamerasrc-ipu6ep;
+
+in
+{
+
+  options.services.v4l2-relayd = {
+
+    enable = mkEnableOption (lib.mdDoc "v4l2-relayd");
```

Multiple instances could work like this:

```suggestion  
    instances = mkOption {
      type = attrsOf (subdmoule {
        options = {
            enable = mkEnableOption (lib.mdDoc "this v4l2-relayd instance");
            
            # ...
        };
      });
      default = {};
      description = mdDoc ''
        Defintion of v4l2-relayd instances.
      '';
    };
```

And we'd then configure `services.v4l2-relayd.instances.ipu6.enable = true;`

---

done